pkg/controller/core/traits/manualscalertrait/manualscalertrait_controller.go

```diff
@@ -65,7 +84,7 @@ func (r *ManualScalerTraitReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	log.Info("Get the manualscaler trait", "ReplicaCount", manualScaler.Spec.ReplicaCount,
-		"WorkloadReference", manualScaler.Spec.WorkloadReference)
+		"Annotations", manualScaler.GetAnnotations())
 
 	// Fetch the workload this trait is referring to
 	var workload unstructured.Unstructured
```

Comment:
encapsulate those code into a function or a method of basic struct is a good way
In that case, core trait or extend trait can use it
Personally I prefer abstract struct, like the following 
 ```
type AbstractTrait struct {
    client.Client
}
func (at *AbstractTrait) getWorkloadByTrait(ctx context.Context, t oam.Trait) (unstructured.Unstructured, error) {
var workload unstructured.Unstructured

	ref := t.GetWorkloadReference()

	workload.SetAPIVersion(ref.APIVersion)
	workload.SetKind(ref.Kind)

	wn := client.ObjectKey{Name: ref.Name, Namespace: t.GetNamespace()}

	if err := at.Get(ctx, wn, &workload); err != nil {
		msg := fmt.Sprintf("can not found trait reference workload, ns:%s,trait:%s,workload:%s", t.GetNamespace(), t.GetName(), ref.Name)
		t.SetConditions(cpv1alpha1.ReconcileError(errors.Wrap(err, msg)))
		return workload, errors.Wrap(err, msg)
	}
	return workload, nil
} 
```

Reply:
sure,  this part of the code can use some refactor

Reply:
Highly recommended!